### DISCUSSION TITLE
Prepare fragments of the upgrading guide for pending breaking changes

### DISCUSSION THREAD

**Discussion by matthewmayer** [faker-js/faker#2521] [Ideas]
I had an idea, as we approach v9 we will have more and more PRs queued up with breaking changes. Then when we decide that the next release will be v9, these will all need to be merged into `next`. At that point we will have to write the upgrading guide https://fakerjs.dev/guide/upgrading.html for upgrading to v9. We could save time on this by adding markdown fragments to each breaking PR which are later assembled into the full upgrading guide.

For example
https://github.com/faker-js/faker/blob/next/docs/guide/upgrading_v9/foo.md in one PR
https://github.com/faker-js/faker/blob/next/docs/guide/upgrading_v9/bar.md in another PR
https://github.com/faker-js/faker/blob/next/docs/guide/upgrading_v9/baz.md in a third PR

This keeps the documentation changes together with the PR itself, allowing them to be easily reviewed.

Then when its time to merge all the PRs, the fragments can simply be concatenated to make the final upgrading.md file.

**Comment by ST-DDT** [comment]
Any suggestion for a naming strategy for those files? e.g. use PR-number?

**Reply by matthewmayer** [reply]
PR number would seem fine to avoid clashes. They are intended to be temporary so it wouldn't matter that much.
